### PR TITLE
improve expand/collapse logic

### PR DESCRIPTION
**Pull request by martinklepsch**
Based on #99.

To Test: 

1. open on mobile device
2. navigate categories, check that the autoscrolling is sensible
3. expand a topic
4. scroll, check that topic stays expanded
5. expand other topic, check previously expanded topic remains expanded

### CODE REVIEW

File: src/open_company_web/components/topic.cljs [hunk 1]
```diff
@@ -61,64 +64,61 @@
...
         (let [section (keyword (:section-name options))
               section-data (:section-data data)
               body (utils/get-topic-body section-data section)]
-          (when (clojure.string/blank? body)
+          (when (and (clojure.string/blank? body) (:force-edit-cb options))
             ((:force-edit-cb options) true)))
         ;; hide the edit button if necessary
-        ((:force-edit-cb options) false)))))
+        (when-let [f (:force-edit-cb options)]
+          (f false))))))
```
Comment (martinklepsch) [inline src/open_company_web/components/topic.cljs]: I can't find any place where `force-edit-cb` is ever set to anything — is this dead code? @bago2k4

Comment (bago2k4) [inline src/open_company_web/components/topic.cljs]: @martinklepsch yep, that was used for mobile editing. I probably missed it when i removed it. good catch

**Comment by bago2k4** [comment]
reviewing

File: src/open_company_web/components/topic.cljs [hunk 2]
```diff
@@ -208,7 +208,9 @@
 
 (defn topic-click [data owner options expanded selected-metric]
   (if (utils/is-mobile)
-    (mobile-topic-animation data owner options expanded)
+    (do (dis/dispatch! [:topic/toggle-expand (keyword (:section-name options))])
+        (mobile-topic-animation data owner options expanded)
+        (after 50 #(scroll-to-topic-top (om/get-ref owner "topic"))))
```
Comment (bago2k4) [inline src/open_company_web/components/topic.cljs]: do we really need another function for this? can't we call directly `(js/setTimeout #(scroll-to-topic-top (om/get-ref owner "topic")) 50)` @martinklepsch

**Comment by martinklepsch** [comment]
@bago2k4 I think the `after` was nice because it was a bit easier to read and the time came first. I guess using it once doesn't justify it's own function but it could have been used in other places as well. Potentially be moved to utils haha.

**Pull request merged**